spec/services/discord/sync_profile_service_spec.rb

```diff
@@ -38,45 +39,95 @@
           discord_configuration[:discord][:server_id],
           user.discord_user_id,
           roles:
             a_collection_containing_exactly(
-              'role_1',
-              'role_2',
-              'role_shared',
-              'role_3',
-              'role_4',
-              'default_role'
+              "role_1",
+              "role_2",
+              "role_shared",
+              "role_3",
+              "role_4",
+              "default_role"
             ),
           nick: user.name
-        )
+        ).and_return(rest_client_double)
 
-        subject.new(user).execute
+        expect { subject.new(user).execute }.to_not(
+          change { AdditionalUserDiscordRole.count }
+        )
       end
     end
 
-    context 'when a user does not have a discord user id' do
+    context "when a user does not have a discord user id" do
       before { user.update(discord_user_id: nil) }
-      it 'does not resets the roles' do
+      it "does not resets the roles" do
         expect(Discordrb::API::Server).not_to receive(:update_member)
 
         subject.new(user).execute
       end
     end
 
-    context 'when configuration is not present' do
+    context "when configuration is not present" do
       before { school.update!(configuration: {}) }
-      it 'does not resets the roles' do
+      it "does not resets the roles" do
         expect(Discordrb::API::Server).not_to receive(:update_member)
 
         subject.new(user).execute
       end
     end
+
+    context "when user is assigned additional discord roles" do
+      before { 4.times { create :discord_role, school: school } }
+
+      let(:additional_discord_role_ids) { DiscordRole.limit(2).pluck(:id) }
+      let(:roles_ids) do
+        DiscordRole.where(id: additional_discord_role_ids).pluck(:discord_id)
+      end
+
+      it "syncs and links user to discord roles" do
+        expect(rest_client_double).to receive(:code).and_return(200)
+        expect(rest_client_double).to receive(:body).and_return(
+          "{\"roles\": [\"role_1\", \"role_2\", \"role_shared\", \"default_role\", \"#{roles_ids.first}\", \"#{roles_ids.last}\"]}"
+        )
+
+        expect(Discordrb::API::Server).to receive(:update_member).with(
+          "Bot #{discord_configuration[:discord][:bot_token]}",
+          discord_configuration[:discord][:server_id],
+          user.discord_user_id,
+          roles:
+            a_collection_containing_exactly(
+              "role_1",
+              "role_2",
+              "role_shared",
+              "role_3",
+              "role_4",
+              "default_role",
+              roles_ids.first,
+              roles_ids.last
+            ),
+          nick: user.name
+        ).and_return(rest_client_double)
+
+        expect {
+          subject.new(
+            user,
+            additional_discord_role_ids: additional_discord_role_ids
+          ).execute
+        }.to change { user.discord_roles.count }.by(2)
```

Comment:
Instead of checking the count change, wouldn't it be more straight-forward to check whether the expected `discord_roles` are there? i.e., `expect(user.discord_roles).to eq(...)`